[PyTorch FE] quantized::linear_relu functionality

Samay10 opened the pull request:
### Details:
 - Implemented quantized::linear_relu by fusing MatMul + Add + ReLU.
 - Integrated the new operation into OpenVINO’s frontend.
 - Added necessary headers and ensured compatibility with existing quantized operations.

### Tickets:
 - Closes #29733
 - Parent Issue: #28584

Samay10 commented:
@slyalin @cavusmustafa Could you please review my PR when you get a chance? Thanks!

mvafin commented on src/core/include/openvino/op/quantized_linear_relu.hpp:
This file is not needed if you define `OP_CONVERTER` in `op_table.cpp`. Please remove the file

mvafin commented on src/frontends/tests/frontend/shared/src/test_quantized_relu.cpp:
These tests are not suitable for operation testing. Please add test in `tests/layer_tests/pytorch_tests`

Samay10 replied:
Sure @mvafin, thanks for clarifying. I will implement the changes right away.

mvafin reviewed (changes requested):
Hello @Samay10 . You haven't implemented the `quantized::linear_relu` operation. Please look into how `quantized::hardswish` is implemented and tested and make your implementation similar

mvafin commented on src/frontends/pytorch/src/op/quantized_linear_relu.cpp:
```diff
@@ -0,0 +1,30 @@
+
+OutputVector translate_quantized_relu(const NodeContext& context) {
+    num_inputs_check(context, 3, 3);
+    
```
```suggestion

```

mvafin commented on src/frontends/pytorch/src/op/quantized_linear_relu.cpp:
```diff
@@ -0,0 +1,30 @@
+#include "openvino/frontend/pytorch/node_context.hpp"
```
Please add copyright in the beginning of the file

Samay10 replied:
hi @mvafin, i have added the copyright

Samay10 commented:
hi @mvafin, I have removed the additional space that was causing 2 checks to fail as guided by you and also added the copyright - it should pass all the checks now. Please check it once, thanks

Samay10 commented on tests/layer_tests/pytorch_tests/test_quantized_linear_relu.py:
```diff
@@ -0,0 +1,46 @@
+        # Quantize input tensor
+        quantized_tensor = torch.quantize_per_tensor(input_tensor, self.scale, self.zero_point, self.dtype)
+        # Apply quantized ReLU operation
+        q_relu = torch.ops.quantized.relu(quantized_tensor)
```
Understood sir - give me some time, I will fix it and push the changes

Samay10 replied:
Hi @mvafin, a couple of questions:
I am experiencing challenges translating PyTorch's quantized linear operations in the OpenVINO PyTorch frontend, specifically with `torch.ops.quantized.linear_pack()`.

## Specific Test Scenario
- Test: Quantized Linear + ReLU Operation
- Configurations Tested:
  - Scales: [1.0, 0.21, 0.62, 0.9999]
  - Zero Points: [0, 4, -7]
  - Data Types: [torch.quint8, torch.qint8]

## Current Error
FAILED test_quantized_linear_relu.py::TestQuantizedLinearReLU::test_quantized_linear_relu
- AttributeError: '_OpNamespace' 'quantized' object has no attribute 'linear_pack'

**How should we handle the `torch.ops.quantized.linear_pack()` representation in the OpenVINO translation?**

Relevant code snippet:
```python
class QuantizedLinearReLU(nn.Module):
    def __init__(self, input_features, output_features, scale, zero_point, dtype):
        # Pre-pack weights
        quantized_weight = torch.quantize_per_tensor(weight, 1.0, 0, torch.qint8)
        self.W_prepack = torch.ops.quantized.linear_pack(quantized_weight, bias)

    def forward(self, input_tensor):
        # Quantized linear + ReLU
        q_output = torch.ops.quantized.linear_relu(
            quantized_input,
            self.W_prepack,
            self.scale,
            self.zero_point
        )
```

mvafin replied:
`AttributeError: '_OpNamespace' 'quantized' object has no attribute 'linear_pack'` is printed by torch, it is not an openvino problem. There is no such operation in torch namespace.

p-wysocki commented:
Hello @Samay10, the code is not properly tested. Due to this fact, combined with a long time of inactivity, we need to reopen the task for other contributors.

If you wish you can keep working on that PR and submit it to the issue if it still open by then.
